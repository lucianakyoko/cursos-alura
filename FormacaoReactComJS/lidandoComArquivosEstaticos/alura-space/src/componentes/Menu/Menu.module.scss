.menu {
  width: 18%;

  &__lista {
    padding-left: 0.5rem;
    margin: 0;
    height: 276px;
    list-style: none;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
  }

  &__item {
    display: flex;
    justify-content: flex-start;
    align-items: center;

    &:first-child {
      a {
        font-weight: 700;
        color: #7b78e5;
        font-family: "GandhiSansRegular";
      }
    }
  }

  img {
    margin-right: 0.5rem;
    width: 25px;
    height: 25px;
    cursor: pointer;
  }

  a {
    text-decoration: none;
    font-size: 1.125rem;
    color: #d9d9d9;
    font-family: "GandhiSansRegular"
  }
}