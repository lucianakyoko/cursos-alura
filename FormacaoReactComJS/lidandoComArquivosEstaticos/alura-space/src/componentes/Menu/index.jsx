import home from '../../assets/icones/home-ativo.png';
import maisCurtidas from '../../assets/icones/mais-curtidas-inativo.png';
import maisVistas from '../../assets/icones/mais-vistas-inativo.png';
import novas from '../../assets/icones/novas-inativo.png';
import surpreendaMe from '../../assets/icones/surpreenda-me-inativo.png';

import styles from './Menu.module.scss';

export default function Menu() {
  return (
    <nav className={styles.menu}>
      <ul className={styles.menu__lista}>
        <li className={styles.menu__item}>
          <img src={home}alt="Ícone home" />
          <a href="/">Início</a>
        </li>
        <li className={styles.menu__item}>
          <img src={maisCurtidas} alt="Ícone mais curtidas" />
          <a href="/">Mais curtidas</a>
        </li>
        <li className={styles.menu__item}>
          <img src={maisVistas} alt="Ícone mais vistas" />
          <a href="/">Mais vistas</a>
        </li>
        <li className={styles.menu__item}>
          <img src={novas} alt="Ícone novas" />
          <a href="/">Novas</a>
        </li>
        <li className={styles.menu__item}>
          <img src={surpreendaMe} alt="Ícone surpreenda-me" />
          <a href="/">Surpreenda-me</a>
        </li>
      </ul>
    </nav>
  )
}
