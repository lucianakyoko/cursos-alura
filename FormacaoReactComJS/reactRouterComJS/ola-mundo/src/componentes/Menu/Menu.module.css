.navegacao {
  height: 112px;
  display: flex;
  justify-content: center;
  align-items: center;
  column-gap: 3rem;
}

@media (max-width: 744px) {
  .navegacao {
      height: 72px;
  }
}