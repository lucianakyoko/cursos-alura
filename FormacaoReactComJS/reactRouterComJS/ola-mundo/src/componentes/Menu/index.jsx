import MenuLink from '../MenuLink';
import styles from './Menu.module.css';

export default function Menu() {
  return (
    <header>
      <nav className={styles.navegacao}>
        <MenuLink to='/'>Inicio</MenuLink>
        <MenuLink to='/sobremim'>Sobre Mim</MenuLink>
      </nav>
    </header>
  );
}