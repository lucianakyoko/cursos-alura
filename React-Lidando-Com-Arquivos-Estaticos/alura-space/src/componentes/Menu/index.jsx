import home from '../../assets/icones/home-ativo.png';
import maisCurtidas from '../../assets/icones/mais-curtidas-inativo.png';
import maisVitidas from '../../assets/icones/mais-vistas-inativo.png';
import novas from '../../assets/icones/novas-inativo.png';
import surpreendaMe from '../../assets/icones/surpreenda-me-inativo.png';

import styles from './Menu.module.scss';

export default function Menu() {
  return (
    <nav className={styles.menu}>
      <ul className={styles.menu__lista}>
        <li className={styles.menu__item}>
          <img src={home} alt="" />
          <a href="/">In√≠cio</a>
        </li>
        <li className={styles.menu__item}>
          <img src={maisCurtidas} alt="" />
          <a href="/">Mais curtidas</a>
        </li>
        <li className={styles.menu__item}>
          <img src={maisVitidas} alt="" />
          <a href="/">Mais vistas</a>
        </li>
        <li className={styles.menu__item}>
          <img src={novas} alt="" />
          <a href="/">Novas</a>
        </li>
        <li className={styles.menu__item}>
          <img src={surpreendaMe} alt="" />
          <a href="/">Surpreenda-me</a>
        </li>
      </ul>
    </nav>
  )
}
