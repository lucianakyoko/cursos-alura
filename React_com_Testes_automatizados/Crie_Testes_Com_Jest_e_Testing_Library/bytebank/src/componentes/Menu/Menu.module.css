.item {
  text-align: center;
}

.menu {
  width: 180px;
  height: 100%;
  border-radius: 8px;
  padding: 24px;
  background-color: #f7f7f7;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  list-style: none;
}

.divisor {
  width: 90%;
  height: 1px;
  background-color: #111111;
  margin: 1rem auto;
}

.link {
  color: #111111;
  font-size: 16px;
  cursor: pointer;
  text-decoration: none;
}

.link:hover {
  color: #000000;
  font-weight: 600;
}
