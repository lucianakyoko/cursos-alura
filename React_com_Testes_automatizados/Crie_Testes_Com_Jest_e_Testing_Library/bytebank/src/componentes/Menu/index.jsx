import React from 'react';
import estilos from './Menu.module.css';

const listaMenu = [
  'Inicial',
  'Transferências',
  'Investimentos',
  'Outros serviços',
];

export default function Menu() {
  return (
    <nav className={estilos.menu}>
      {listaMenu.map((item, indice) => {
        return (
          <div key={item} className={estilos.item}>
            <a href="/" className={estilos.link}>
              {item}
            </a>
            {indice !== listaMenu.length - 1 && (
              <div className={estilos.divisor} />
            )}
          </div>
        );
      })}
    </nav>
  );
}
