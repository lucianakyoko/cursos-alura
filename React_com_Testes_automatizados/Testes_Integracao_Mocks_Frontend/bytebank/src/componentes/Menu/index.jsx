import React from 'react';
import { Link } from 'react-router-dom';
import estilos from './Menu.module.css';

const listaMenu = [
  { link: 'Início', href: '/' },
  { link: 'Cartões', href: '/cartoes' },
  { link: 'Serviços', href: '/servicos' },
  { link: 'Investimentos', href: '/investimentos' },
];

export default function Menu({ path }) {
  return (
    <nav className={estilos.menu}>
      {listaMenu.map((item, indice) => {
        return (
          <div key={item.href} className={estilos.item}>
            <Link
              to={item.href}
              className={`${estilos.link} ${
                path === item.href && estilos.linkAtivo
              }`}
            >
              {item.link}
            </Link>
            {indice !== listaMenu.length - 1 && (
              <div className={estilos.divisor} />
            )}
          </div>
        );
      })}
    </nav>
  );
}
