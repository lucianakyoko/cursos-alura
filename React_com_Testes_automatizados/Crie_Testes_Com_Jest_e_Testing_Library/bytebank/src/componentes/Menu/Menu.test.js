import { render, screen } from '@testing-library/react';
import Menu from './index';

test('Deve renderizar um link para a página inicial', () => {
  render(<Menu />);
  const linkPaginaInicial = screen.getByText('Inicial');
  expect(linkPaginaInicial).toBeInTheDocument();
});

test('Deve renderizar uma lista de links', () => {
  render(<Menu />);
  const listaDeLinks = screen.getAllByRole('link');
  expect(listaDeLinks).toHaveLength(4);
});

test('Não deve renderizar uo link para Extrato', () => {
  render(<Menu />);
  const linkExtrato = screen.queryByText('Extrato');
  expect(linkExtrato).not.toBeInTheDocument();
});

test('Não deve renderizar uo link para Extrato', () => {
  render(<Menu />);
  const linkExtrato = screen.queryByText('Extrato');
  expect(linkExtrato).not.toBeInTheDocument();
});

test('eve renderizar uma lista de links com a classe link',  () => { 
  render(<Menu />)
  const links = screen.getAllByRole('link')
  links.forEach(link => expect(link).toHaveClass('link'))
  expect(links).toMatchSnapshot()
 })

