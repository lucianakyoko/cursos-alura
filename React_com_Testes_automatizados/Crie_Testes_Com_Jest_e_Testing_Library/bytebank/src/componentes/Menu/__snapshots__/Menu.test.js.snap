// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eve renderizar uma lista de links com a classe link 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="link"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
